_sips/sips/2019-07-27-binary-integer-literals.md

```diff
@@ -0,0 +1,31 @@
+---
+layout: sip
+discourse: true
+title: SIP-NN - Support Binary Integer Literals
+---
+
+**By: NthPortal**
+
+## History
+
+| Date          | Version                  |
+|---------------|--------------------------|
+| Jul 27th 2019 | Initial Draft            |
+
+Your feedback is welcome! If you're interested in discussing this proposal, head over to [this](https://contributors.scala-lang.org/t/pre-sip-binary-literals/3559) Scala Contributors thread and let me know what you think.
+
+## Proposal
+
+Support binary integer literals. For example, the binary literal `0b00101010` would have the integer value `42`.
+
+## Motivation
+
+Several other major languages support binary integer literals, including [Java](https://docs.oracle.com/javase/specs/jls/se12/html/jls-3.html#jls-3.10.1) (as of [Java 7](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.10.1)), [Python](https://docs.python.org/3/reference/lexical_analysis.html#integer-literals) and [Rust](https://doc.rust-lang.org/stable/reference/tokens.html#number-literals).
```

Comment:
I would also mention [Kotlin](https://kotlinlang.org/docs/reference/basic-types.html#literal-constants), since it is close to Scala.

Reply:
added